src/Adapter/Manufacturer/CommandHandler/AbstractManufacturerCommandHandler.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Manufacturer\CommandHandler;
+
+use Manufacturer;
+use PrestaShop\PrestaShop\Adapter\Manufacturer\AbstractManufacturerHandler;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\ManufacturerException;
+use PrestaShopException;
+
+/**
+ * Provides reusable methods for manufacturer command handlers
+ */
+abstract class AbstractManufacturerCommandHandler extends AbstractManufacturerHandler
+{
+    /**
+     * Deletes legacy Manufacturer
+     *
+     * @param Manufacturer $manufacturer
+     *
+     * @return bool
+     *
+     * @throws ManufacturerException
+     */
+    protected function deleteManufacturer(Manufacturer $manufacturer)
```

Comment:
This is a question for Product Team 😉 @colinegin @MatShir

Reply:
Again, sorry for the delay, I guess I missed your ping ! And again, do not hesitate to add the "waiting for PM" label, it's easier for us to review your questions quickly.

About your question : 
In my opinion, it does not make sense to create an address if there is no associated brand.
BUT, to be sure there's no hidden dependency, I would rather choose to enable creation / edition of an address without any associated manufacturer. 
This means that in the manufacturer dropdown, "no manufacturer" should be an option as well as the exisiting brands.

What do you think @matks and @zuk3975 ?

Reply:
All right for me 🤝

Reply:
@colinegin if I understand correctly, we are changing behavior, right?

Before it worked like so:
* If there are manufacturers, then user can choose one from the drop-down select (by default 1st is preselected), thus not allowing user to create address without manufacturer (which makes sense).
* If there are no manufacturers, you can still create address, but without manufacturer, as there no choices in the drop-down (which is weird behavior).

With your suggestion, we would add "No manufacturer" option even when manufacturers are available in drop-down select, thus allowing user to create address without manufacturer which does not seem logical, wdyt?

Reply:
If you delete manufacturer, the associated address will end up being without it anyway.
So as far as I understand we are just adapting front-end with colinegin solution by allowing to control this in edit/create.

Reply:
Yes, I agree that it does not make any sense, but i'm afraid we could break things if for now we make it impossible to create / edit an address with no manufacturer. Wdyt ?
So my recommendation would be to enable the same options on any scenario (create, edit, delete, no manufacturer available, etc.)

Reply:
>  but i'm afraid we could break things if for now

We wouldn't break, if we keep legacy behavior. This means address can be created without manufacturer **only when** (this is legacy behavior) no manufacturers are available. Otherwise, we are adding option for user to do things wrong in yet another way. :/